Sources/Apollo/NormalizedCache.swift

```diff
@@ -1,5 +1,46 @@
 import Foundation
 
+/// An object that drives behavior of clear operations on a given cache.
+public struct CacheClearingPolicy: Equatable {
```

Comment:
Can you explain why this struct exists rather than just exposing the `Policy` enum directly, since they seem almost identical?

Reply:
I knew this would come up, and was preparing for this. We can discussion synchronously outside of this PR if you want about it 😄 

I admit that this isn't the most elegant implementation, as I've seen some other, but it is the simplest for what I was hoping to help achieve with it.

This is a pattern that I've seen widely accepted and adopted in the Server-side Swift realm that assists in giving library evolution to SwiftPM static libraries, as adding new cases are technically breaking changes.

When you use a enum-like-struct new cases can arise and be added without much worry for your users. Dave DeLong tweeted about it a few weeks ago: https://twitter.com/davedelong/status/1319368474444005376

A blog post on the anti-pattern nature of raw enums: https://www.jessesquires.com/blog/2016/07/31/enums-as-configs/

A lot of this can also be found in this forum discussion: https://forums.swift.org/t/extensible-enumerations-for-non-resilient-libraries/35900

Reply:
Super-hard disagree on this one. I would _much_ rather people have to consciously handle any additional cases (especially when most of the handling is going to be internal to the library) than have default behavior which doesn't throw a warning when a new case is added. In my opinion the breaking behavior of this is a feature, not a bug. 

Enums certainly aren't the solution for everything, but in this case where we have: 

- A fixed set of cases we're handling
- Not allowing the user to create subclasses which can do their own handling
- Not adding any additional properties that can't be set up as an associated object
- Using this as a way to tell the internals of our library what to do rather than as a way to say "Here are the options of what this could be" to an external library

an `enum` is absolutely what we want here.

Reply:
This is fair, and just wanted to do it this way first to see the discussion 👍 I've made the change to a plain enum